Fail early when a JDBS driver is not in the classpath

## Background
Users are getting:
```
Caused by: java.sql.SQLException: No suitable driver
```

This usually means the driver is not at the classpath. Pramen can pro-actively check this and fail earlier, especially from the `SparkSink`.

## Feature
Fail early when a JDBS driver is not in the classpath.

## Example [Optional]
--

## Proposed Solution [Optional]
--

--- a/pramen/core/src/main/scala/za/co/absa/pramen/core/sink/SparkSink.scala
+++ b/pramen/core/src/main/scala/za/co/absa/pramen/core/sink/SparkSink.scala
@@ -241,9 +241,32 @@ object SparkSink extends ExternalChannelFactory[SparkSink] {
   val FILE_FORMATS: Seq[String] = Seq("csv", "json", "parquet", "avro", "orc", "xml")
 
   override def apply(conf: Config, parentPath: String, spark: SparkSession): SparkSink = {
+    val extraOptions = ConfigUtils.getExtraOptions(conf, "option")
+
+    val format = ConfigUtils.getOptionString(conf, FORMAT_KEY).getOrElse(DEFAULT_FORMAT)
+
+    if (format.equalsIgnoreCase("jdbc")) {
+      // Initialize JDBC driver
+      if (extraOptions.contains("driver")) {
+        val driver = extraOptions("driver")
+        try {
+          log.info(s"Initializing JDBC driver: $driver")
+          Class.forName(driver)
+        } catch {
+          case ex: Throwable => log.warn(s"Unable to initialize the JDBC driver: '$driver'. Still trying using Spark...", ex)
+        }
+      } else {
+        throw new IllegalArgumentException(s"Mandatory SparkSink JDBC option 'option.driver' is not defined at '$parentPath'.")
+      }
+
+      if (!extraOptions.contains("url")) {
+        throw new IllegalArgumentException(s"Mandatory SparkSink JDBC option 'option.url' is not defined at '$parentPath'.")
+      }
+    }
+
     new SparkSink(
-      ConfigUtils.getOptionString(conf, FORMAT_KEY).getOrElse(DEFAULT_FORMAT),
-      ConfigUtils.getExtraOptions(conf, "option"),
+      format,
+      extraOptions,
       ConfigUtils.getOptionString(conf, MODE_KEY).getOrElse(DEFAULT_MODE),
       ConfigUtils.getOptListStrings(conf, PARTITION_BY_KEY),
       ConfigUtils.getOptionInt(conf, NUMBER_OF_PARTITIONS_KEY),

--- a/pramen/core/src/test/scala/za/co/absa/pramen/core/sink/SparkSinkSuite.scala
+++ b/pramen/core/src/test/scala/za/co/absa/pramen/core/sink/SparkSinkSuite.scala
@@ -47,7 +47,7 @@ class SparkSinkSuite extends AnyWordSpec with SparkTestBase with TempDirFixture 
           |  partition.by = [ info_date ]
           |  save.empty = false
           |
-          |  options {
+          |  option {
           |    compression = gzip
           |  }
           |""".stripMargin
@@ -55,6 +55,60 @@ class SparkSinkSuite extends AnyWordSpec with SparkTestBase with TempDirFixture 
       val sink = SparkSink(conf, "parent", spark)
 
       assert(sink.isInstanceOf[SparkSink])
+    }
+
+    "construct a JDBC sink from config" in {
+      val conf = ConfigFactory.parseString(
+        """  format = "jdbc"
+          |  mode = "overwrite"
+          |  records.per.partition = 1
+          |
+          |  option {
+          |    driver = "org.hsqldb.jdbc.JDBCDriver"
+          |    url = "jdbc:hsqldb:mem:mydb;sql.enforce_size=false"
+          |  }
+          |""".stripMargin
+      )
+
+      val sink = SparkSink(conf, "parent", spark)
+
+      assert(sink.isInstanceOf[SparkSink])
+    }
+
+    "throw an exception if mandatory driver option is missing" in {
+      val conf = ConfigFactory.parseString(
+        """  format = "jdbc"
+          |  mode = "overwrite"
+          |  records.per.partition = 1
+          |
+          |  option {
+          |    url = "jdbc:hsqldb:mem:mydb;sql.enforce_size=false"
+          |  }
+          |""".stripMargin
+      )
+      val ex = intercept[IllegalArgumentException] {
+        SparkSink(conf, "parent", spark)
+      }
+
+      assert(ex.getMessage == "Mandatory SparkSink JDBC option 'option.driver' is not defined at 'parent'.")
+    }
+
+    "throw an exception if mandatory url option is missing" in {
+      val conf = ConfigFactory.parseString(
+        """  format = "jdbc"
+          |  mode = "overwrite"
+          |  records.per.partition = 1
+          |
+          |  option {
+          |    driver = "org.hsqldb.jdbc.JDBCDriver"
+          |  }
+          |""".stripMargin
+      )
+      val ex = intercept[IllegalArgumentException] {
+        SparkSink(conf, "parent", spark)
+      }
+
+      assert(ex.getMessage == "Mandatory SparkSink JDBC option 'option.url' is not defined at 'parent'.")
     }
   }
 

